assets/js/modules/adsense/util/status.js

```diff
@@ -0,0 +1,246 @@
+ * Status utlities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const ACCOUNT_STATUS_NONE = 'none';
+export const ACCOUNT_STATUS_DISAPPROVED = 'disapproved';
+export const ACCOUNT_STATUS_GRAYLISTED = 'graylisted';
+export const ACCOUNT_STATUS_MULTIPLE = 'multiple';
+export const ACCOUNT_STATUS_NO_CLIENT = 'no-client';
+export const ACCOUNT_STATUS_PENDING = 'pending';
+export const ACCOUNT_STATUS_APPROVED = 'approved';
+
+// TODO: If AdSense API exposes more information, site status can be determined
+// much more accurately than the two statuses below. See also function docs.
+export const SITE_STATUS_NONE = 'none';
+export const SITE_STATUS_ADDED = 'added';
+
+/**
+ * Determines the AdSense account status for given input data.
+ *
+ * This utility function should be used in combination with data retrieved from
+ * the datastore, hence passing undefined (loading state) is supported.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}  data               Input data to determine account status.
+ * @param {?Array}  data.accounts      List of account objects retrieved from the API.
+ * @param {?Array}  data.clients       List of client objects retrieved from the API.
+ * @param {?Array}  data.alerts        List of alert objects retrieved from the API.
+ * @param {?Object} data.error         Error object if one of the API requests failed.
+ * @param {?string} data.prevAccountID Account ID, if already known from before.
+ * @param {?string} data.prevClientID  Client ID, if already known from before.
+ * @return {?string} Account status determined, or undefined if one of the required
+ *                   parameters is undefined.
+ */
+export const determineAccountStatus = ( {
+	accounts,
+	clients,
+	alerts,
+	error,
+	prevAccountID,
+	prevClientID,
+} ) => {
+	if ( 'undefined' === typeof accounts || 'undefined' === typeof prevAccountID ) {
+		return errorToStatus( error );
+	}
+
+	const accountID = determineAccountID( { accounts, prevAccountID } );
+	if ( ! accountID ) {
+		// If there are accounts, but the account ID cannot be determined, it
+		// means that there are multiple accounts and the user needs to select
+		// one.
+		if ( accounts.length ) {
+			return ACCOUNT_STATUS_MULTIPLE;
+		}
+		return ACCOUNT_STATUS_NONE;
+	}
+
+	if ( 'undefined' === typeof alerts ) {
+		return errorToStatus( error );
+	}
+
+	const hasGraylistedAlert = !! alerts.filter( ( alert ) => {
```

Comment:
Slightly more efficient here would be [`Array.some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some):

```js
alerts.return( ( alert ) => {
	return 'GRAYLISTED_PUBLISHER' === alert.type;
} );
```

It will return `true` as soon as it finds a matching element, so it won't iterate over all the alerts. Probably minor here, but 🤷‍♂

Reply:
Thanks, I wasn't aware that existed!